Tech@HubAPI/Tech@HubAPI/Forms/ChangePasswordForm.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Linq;
+using Tech_HubAPI.Services;
+using Tech_HubAPI.Models;
+
+public class ChangePasswordForm
+{
+    public ChangePasswordForm(string userName, string oldPassword, string newPassword, string newPasswordRetyped)
+    {
+            this.UserName = userName;
+            this.OldPassword = oldPassword;
+            this.NewPassword = newPassword;
+            this.NewPasswordRetyped = newPasswordRetyped;
+    }
+
+        public string UserName { get; set; }
+        public string OldPassword { get; set; }
+        public string NewPassword { get; set; }
+        public string NewPasswordRetyped { get; set; }
+
+        public Boolean checkNewPassword(string newPassword, string newPasswordRe )
```

Comment:
Instead of the ```checkNewPassword``` method, can you implement a ```Validate``` method that looks like this:

```
public void Validate()
{
   if (NewPassword != NewPasswordRetyped)
   {
      throw new ArgumentException("The new and retyped passwords do not match.");
   }
}
```

This is a pattern we have been using to do validation checks on the other forms. In the endpoint, you can just call
```
try
{
   passwordForm.Validate();
}
catch (ArgumentException ex)
{
   return Conflict(ex.Message);
}
```

Or if the ```Conflict``` response doesn't let you give it a message, you could return
```
return BadRequest(ex.Message);
```